### ISSUE TITLE
monitored_metrics error

### ISSUE THREAD

**Issue by preekatz** [idealo/image-super-resolution#18]
I got an error while training the network as follows:
TypeError(): train() missing 1 required positional argument: 'monitored metrics'

**Comment by preekatz** [comment]
Then when i set monitered_metrics='accuracy' this error is solved but i get another error regarding weights. PFA screenshot for your reference.
![error](https://user-images.githubusercontent.com/48949016/55909181-f90e6d00-5bf8-11e9-8f7e-dc6bbc561171.png)

**Comment by preekatz** [comment]
I am trying to use the model for transfer learning. I wanted to know what should be values of 'C','D','G','G0','T' in the following code:
`rrdn  = RRDN(arch_params={'C':3, 'D':4, 'G':64, 'G0':64, 'T':9, 'x':scale}, patch_size=lr_train_patch_size)
`
To load weight file 'rdn-C6-D20-G64-G064-x2_ArtefactCancelling_epoch219.hdf5'

**Comment by preekatz** [comment]
![error](https://user-images.githubusercontent.com/48949016/55944378-1decf980-5c66-11e9-9774-d799fb46b4de.png)

**Comment by preekatz** [comment]
![error1](https://user-images.githubusercontent.com/48949016/55944489-57256980-5c66-11e9-8bf8-24811c905e16.jpg)

**Comment by cfrancesco** [comment]
> Then when i set monitered_metrics='accuracy' this error is solved but i get another error regarding weights. PFA screenshot for your reference.

In the documentation you can read about the parameters needed for `Trainer.train()` :
https://idealo.github.io/image-super-resolution/train/trainer/
What worries me is that you could give a string instead of a key value pair for `monitored_metrics`. Could you train without errors?

> I am trying to use the model for transfer learning. I wanted to know what should be values of 'C','D','G','G0','T' in the following code:
rrdn = RRDN(arch_params={'C':3, 'D':4, 'G':64, 'G0':64, 'T':9, 'x':scale}, patch_size=lr_train_patch_size) 
To load weight file 'rdn-C6-D20-G64-G064-x2_ArtefactCancelling_epoch219.hdf5'

As for the weights, interesting, I never tried to use the pre-trained weights of the RDN to kick start a RRDN model. You could use the flag `by_name=True` in `load_weights`, but the issue here would be that RDN and RRDN actually share very few layers names. To bypass this you can try to manually iterate through the layers and extract weights matrices from the pre-trained weights and assigning them to the individual layers of the RRDN model.
There is no 1-to-1 correspondence obviously between the two models, so there is no correct values of 'C','D','G','G0','T',  however, you could divide the 20 RDB from RDN into any combination divisors of 20 T RRDB (such as T=5, D=4 or T=10, D=2) and keep C=6. I'm not sure if this will give any benefit, but it might be better than random initialization.

For your last screenshots, it seems that the scripts are receiving an invalid path. Could you post what pre-trained weights path are you giving the Trainer? Does it contain the double `\` as in your screenshots? Unfortunately I have no way of testing this in a Windows environment at the moment..

**Comment by preekatz** [comment]
Hello,

Actually we have computation resources issues which is why i wanted to
further train your pretrained network for our text image data set. I wanted
to enhance Low resolution image by factor of 2. Currently i am using
rdn-C6-D20-G64-G064-x2_ArtefactCancelling_epoch216.hdf5 into RDN model with
C:6 D:20 G:64 G0:64
i did change the code where weights and logs are saved and its working.

**Comment by preekatz** [comment]
Also our data set consist of around 25,000 image pairs and training from
scratch on such a small data with 16lakhs parameters model will cause
overfitting.

**Comment by cfrancesco** [comment]
25000 images is a very large dataset for ISR. Beware that quality, rather than quantity, is key. The model we provide was trained on the DIV2K dataset, which contains 800 training images. The main reason is that from each image thousands of different patches can be extracted. Please close the issue if your problem was solved.
